code/modules/events/immovable_rod.dm

```diff
@@ -90,10 +90,17 @@ In my current plan for it, 'solid' will be defined as anything with density == 1
 
 /obj/effect/immovablerod/event
 	var/tiles_moved = 0
+	// The base chance to "damage" the floor when passing. This is not guaranteed to cause a full on hull breach.
+	// Chance to expose the tile to space is like 60% of this value.
+	var/floor_rip_chance = 40
 
 /obj/effect/immovablerod/event/Move()
 	var/atom/oldloc = loc
 	. = ..()
 	tiles_moved++
+	if(prob(floor_rip_chance))
+		var/turf/simulated/floor/T = get_turf(oldloc)
+		if(istype(T))
+			T.ex_act(2)
```

Comment:
```suggestion
			T.ex_act(EXPLODE_HEAVY)
```
we have defines for these.

Reply:
Was looking at the `.ex_act(2)` above when writing this, so was not aware of the constants existance 🙃 
Fixed, thanks!